creator-node/src/components/tracks/tracksComponentService.js

```diff
@@ -61,6 +64,80 @@ const handleTrackContentRoute = async ({ logContext }, requestProps) => {
   return resp
 }
 
+async function handleTranscodeAndSegment(
+  { logContext },
+  { fileName, fileDir }
+) {
+  return TrackContentUploadManager.transcodeAndSegment(
+    { logContext },
+    { fileName, fileDir }
+  )
+}
+
+async function handleTrackHandOff({ logContext }, requestProps) {
+  const {
+    fileName,
+    fileDir,
+    fileDestination,
+    session,
+    libs,
+    AsyncProcessingQueue
+  } = requestProps
+  const { cnodeUserUUID } = session
+  const logger = genericLogger.child(logContext)
+
+  const codeBlockTimeStart = getStartTime()
+
+  // If the Content Ndoe is not initialized yet, the spID will not
```

Comment:
move this out into the flow step above where you decide whether the primary will handle or not - by the time you enter into this part of the flow, you should already KNOW that you are definitely going to hand off. this will simplify the branching logic